webapp/src/reducers/index.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+import {combineReducers} from 'redux';
+
+import ActionTypes from '../action_types';
+
+function connected(state = false, action) {
+    switch (action.type) {
+    case ActionTypes.AUTH_CONNECTED:
+      return action.data;
+    default:
+      return state;
+    }
+}
+
+function disconnected(state = true, action) {
```

Comment:
Should be disconnect just `!connected`? is needed a separate reducer? if we have different states for connected, can be disconnected a selector, instead of another reducer?, should be the same for `authenticated`? Looks like are all handling the same data, and they are just views of that data.

Reply:
Removed `disconnected` and `connected` as they are redundant like you mention.  It's only `authenticated` now.